File: archive/split-a-string-in-balanced-strings.js
const balancedStringSplit = function(s) {
  let ans = 0
  let balans = 0
  s.split('').forEach(l => {
    if (l === 'R') {balans+=1} else {balans-=1}
    if (balans === 0) {ans++}
  })

  return ans
}

File: go-archive/split-a-string-in-balanced-strings.go
func balancedStringSplit(s string) int {
    arr := strings.Split(s, ""); ans:=0; b:=0;
    for i:=0; i < len(arr); i++ {
        if arr[i] == "R" {b++} else {b--}
        if b == 0 {ans++}
    }

    return ans
}